lib/infrastructure/class-wp-json-server.php

```diff
@@ -638,6 +638,19 @@ public function dispatch( $request ) {
 					$response = json_ensure_response( $response );
 				}
 
+				$data = $response->get_data();
+				$properties = ! empty( $handler['schema']['properties'] ) ? $handler['schema']['properties'] : array();
+				foreach( $properties as $key => $field_args ) {
+					if ( ! isset( $field_args['context'] ) ) {
+						continue;
+					}
+					$context = (array) $field_args['context'];
+					if ( ! in_array( $request['context'], $context ) ) {
```

Comment:
There's a problem with this code in that, for create / update requests, the `$request['context']` is set within the controller's method, and not exposed back to the server:

```
NULL
status
array (
  0 => 'embed',
  1 => 'view',
  2 => 'edit',
)
NULL
type
array (
  0 => 'embed',
  1 => 'view',
  2 => 'edit',
)
```

For create / update requests, we should probably have `context=edit` by default. And it would be nice to get some final context from the response object.